Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,358 @@
+                            };
+                        }
+                        else
+                        {
+                            result.Right = new ConstantNode()
+                            {
+                                Value = constantExpression.Value,
+                                Type = GetKind(constantExpression.Value.GetType())
+                            };
+                        }
+                    }
+                }
+                else
+                {
+                    throw new NotSupportedException(node + " is not a supported a supported string expression.");
+                }
+
+                returnNode = result;
+            }
+            else
+            {
+                throw new NotSupportedException(node.Method + " is not a supported method.");
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitBinary(BinaryExpression binaryExpression)
+        {
+            ExpressionNode returnNode;
+            if (binaryExpression.NodeType == ExpressionType.AndAlso)
+            {
+                var andNode = new AndNode();
+                andNode.Left = Extract(binaryExpression.Left);
+                andNode.Right = Extract(binaryExpression.Right);
+                returnNode = andNode;
+            }
+            else if (binaryExpression.NodeType == ExpressionType.OrElse)
+            {
+                var orNode = new OrNode();
+                orNode.Left = Extract(binaryExpression.Left);
+                orNode.Right = Extract(binaryExpression.Right);
+                returnNode = orNode;
+            }
+            else
+            {
+                ComparisonNode comparisonNode;
+                switch (binaryExpression.NodeType)
+                {
+                    case ExpressionType.Equal:
+                        comparisonNode = new EqualityNode();
+                        break;
+                    case ExpressionType.NotEqual:
+                        comparisonNode = new NotEqualNode();
+                        break;
+                    case ExpressionType.LessThan:
+                        comparisonNode = new LtNode();
+                        break;
+                    case ExpressionType.LessThanOrEqual:
+                        comparisonNode = new LteNode();
+                        break;
+                    case ExpressionType.GreaterThan:
+                        comparisonNode = new GtNode();
+                        break;
+                    case ExpressionType.GreaterThanOrEqual:
+                        comparisonNode = new GteNode();
+                        break;
+                    default:
+                        throw new NotSupportedException($"The operator '{binaryExpression.NodeType}' is not supported");
+                }
+
+                if (binaryExpression.Left is MemberExpression binaryLeftMemberExpression)
```

Comment:
You have a lot of repeated code here, try to avoid this. There is no difference between left and right here